### DISCUSSION TITLE
Remote Database Access

### DISCUSSION THREAD

**Discussion by StarGeezerPhil** [aaronwmorris/indi-allsky#1005] [Q&A]
I'm sure I remember @aaronwmorris talking about the potential of moving to MySQL, which would make this question redundant (but searching I can't find any reference to this, so might just be imagining it).

As a sqlite3 db the current database is aimed to be used on local storage, simply reading the db file as/when needed.

I'd like to remotely access the db (read-only) in order to carry out some analysis of the data and also perform some basic functions (like emailing myself a reference of the best shots from the night for example - or key metrics when I'm in the middle of tweaking settings).

I've already shared the db files via samba, which allows me to access the db on my windows box using BeeKeeper for instance.

However, when I try to access this share via another raspberry pi I'm having no success.

Would anyone be able to offer any advice as to how to access a remote sqlite db file from one Pi to another Pi?

**Comment by aaronwmorris** [comment]
NFS or SMB access to a sqlite database will _probably_ work fine as long as the WAL journal pragma is enabled for the connections.

```
PRAGMA journal_mode=WAL;
```

There is one major caveat.  I do not expect it to work very well across platforms.  If the database is created on a Raspberry Pi (arm), I do not think a Windows machine (x86_64) would be able to access the database (or vice versa).  I am not even sure a Raspi in 32-bit mode could access the DB on a 64-bit Raspi.  The sqlite database is a binary format that is platform specific.  You can convert the database by dumping and recreating on the destination platform.

I do have a script I wrote to convert the sqlite database to a mysql database.  indi-allsky will run fine with a mysql database.  I have not tested the script in a while, but it should work.

https://github.com/aaronwmorris/indi-allsky/wiki/Database-conversion

**Reply by StarGeezerPhil** [reply]
Sorry I'm not familiar with what your first suggestion was.

The sqlite db works fine on Windows using BeeKeeper to directly access the file on a share (via samba).

I tried to recreate the same sharing between the two Pi devices, but forgot samba was for windows - so reshared using a native protocol (forgotten which atm) - can see the files fine and (through PHP) appear to successfully connect, but appears to fail to retrieve any data:

Fatal error: Uncaught Error: Call to a member function fetchArray() on bool

(But, only thrown a couple of minutes into this, so might just be me!!)

With the database conversion - would this update indi-allsky to always use the MySQL or is this a one-off / adhoc conversation that would not stay live... and would it pose problems during future updates?

**Reply by aaronwmorris** [reply]
You should be able to mount a SMB share in Linux.  I think you have to install the `cifs-utils` package in Debian.

```
sudo mount -t cifs -o username=username \\10.1.2.3\share /mnt/folder
```

If you perform the mysql conversion, yes, you can configure indi-allsky to use the new mysql database and stop using sqlite entirely.

https://github.com/aaronwmorris/indi-allsky/wiki/MySQL-MariaDB-Information
